src/lib.rs

```diff
@@ -62,6 +63,28 @@ pub enum RunError<E> {
     TimedOut,
 }
 
+impl<E> fmt::Display for RunError<E>
+    where E: error::Error + 'static
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            RunError::User(ref err) => fmt::Display::fmt(&err, f),
```

Comment:
Just a suggestion but this looks cleaner in my book.

```suggestion
            RunError::User(ref err) => err.fmt(f),
```